go/teams/chain.go

```diff
@@ -347,175 +347,116 @@ func NewTeamSigChainPlayerWithState(g *libkb.GlobalContext, reader keybase1.User
 	return res
 }
 
+// Get the latest state.
+// The caller may _not_ modify the returned state.
 func (t *TeamSigChainPlayer) GetState() (res TeamSigChainState, err error) {
 	t.Lock()
 	defer t.Unlock()
 
 	if t.storedState != nil {
-		// The caller shouldn't modify the returned value, but that's really easy to screw up
-		// so DeepCopy to defend our internal state.
-		return t.storedState.DeepCopy(), nil
+		return *t.storedState, nil
```

Comment:
This used to happen once per batch of links but now it happens once per link. So this is some premature optimization. A deep copy could defend against incorrect code downstream. What do you think?